Add error handling

DeLuks2006 opened the pull request:
Added Error Messages to `wm.c`.
Please let me know if there is something to fix!

P.S. I'm going to try to create the keyboard navigation (and some shortcuts) too :)

DeLuks2006 committed:
add error handling

nir9 commented on wm.c:
```diff
@@ -1,20 +1,37 @@
     HMODULE wmDll = LoadLibraryW(L"wm_dll");
+
+    if (wmDll == NULL) {
+        wprintf( L"[!] Error while loading library.\n Error: %ld", GetLastError());
```
Thanks for the contribution! Could you also return an error?

nir9 commented on wm.c:
```diff
@@ -1,20 +1,37 @@
     FARPROC shellProc = GetProcAddress(wmDll, "ShellProc");
 
+    if (shellProc == NULL) {
+        wprintf(L"[!] Error while getting process address.\nError: %ld", GetLastError());
```
Here I would recommend using something like goto cleanup since we want to free the DLL and return with an error both here and in the next failure check of hookHandle - you can read about the goto cleanup pattern on the internet, its pretty much the only good use for goto in c in my opinion :)

nir9 commented on wm.c:
```diff
@@ -1,20 +1,37 @@
+
     hookHandle = SetWindowsHookExW(WH_SHELL, shellProc, wmDll, 0);
 
+    if (hookHandle = NULL) {
```
typo: = should be ==

DeLuks2006 committed:
Add Error Handling

DeLuks2006 committed:
Add Error Handling

DeLuks2006 committed:
Delete shortcuts.c

DeLuks2006 committed:
Delete shortcuts.h

DeLuks2006 committed:
Update .gitignore

DeLuks2006 committed:
Merge branch 'nir9:master' into master

DeLuks2006 commented:
Okay... Let me know if this is better! 
I added a cleanup, fixed the typo and added a return `EXIT_FAILURE` and `EXIT_SUCCESS`

(Don't mind the random deletion of shortcuts.c and shortcuts.h its not done yet but I accidentaly pushed it too 😅)

nir9 commented on wm.c:
```diff
@@ -1,21 +1,54 @@
+    Sleep(INFINITE);
+
+    cleanup:
+        if (shellProc) {
+            FreeLibrary(shellProc);
```
```suggestion
        if (wmDll) {
            FreeLibrary(wmDll);
```

You are passing the procedure instead of the DLL by mistake

DeLuks2006 committed:
Update wm.c

DeLuks2006 commented:
That should be it then. Thank you very much for the reviews!

nir9 committed:
Merge branch 'master' into master

nir9 commented:
Merged, thanks!
